Makefile

```
@@ -1,12 +1,22 @@
+TEST_COUNT ?= 10
+COVERAGE_FILE ?= coverage.out
 
-ensure_go_1_21:
+build: ensure_go_version
+	go build -v ./...
+
+generate: ensure_go_version
+	go install github.com/vektra/mockery/v2@v2.43.0
+	mockery
+
+test: ensure_go_version
+	go test -race -fullpath -shuffle on -count $(TEST_COUNT) -coverprofile=$(COVERAGE_FILE) ./...
+
+lint: ensure_go_version
+	golangci-lint run -c .golangci.yml
```

I recommend making this

```suggestion
	golangci-lint run --tests=true --verbose --max-issues-per-linter 0 --max-same-issues 0
```

So that the lint run can catch everything in one go.

---

Nice. Updated .golangci.yml to include these 3 options.